src/marten/db/query/sql/concerns/sanitizer.cr

```diff
@@ -0,0 +1,67 @@
+module Marten
+  module DB
+    module Query
+      module SQL
+        module Sanitizer
+          private NAMED_PARAMETER_RE        = /(:?):([a-zA-Z]\w*)/
+          private POSITIONAL_PARAMETER_CHAR = '?'
+          private POSITIONAL_PARAMETER_RE   = /#{"\\" + POSITIONAL_PARAMETER_CHAR}/
+
+          private def sanitize_positional_parameters(
+            query : String,
+            params : Array(::DB::Any),
+            connection : Connection::Base? = nil
+          )
+            if query.count(POSITIONAL_PARAMETER_CHAR) != params.size
+              raise Errors::UnmetQuerySetCondition.new("Wrong number of parameters provided for query: #{query}")
+            end
+
+            parameter_offset = 0
+            sanitized_query = query.gsub(POSITIONAL_PARAMETER_RE) do
+              # Conditional placeholder generation
+              if connection
+                parameter_offset += 1
+                connection.parameter_id_for_ordered_argument(parameter_offset)
+              else
+                "%s"
+              end
+            end
+
+            {sanitized_query, params}
+          end
+
+          private def sanitize_named_parameters(
```

Comment:
Same question here: this method should be public and unit-tested ideally. The fact that we created this module also means that it must now be tested. 🙂